src/atomic/atomic.lisp

```diff
@@ -36,18 +44,20 @@
   cell)
 
 (defun make-atomic-integer (&key (value 0))
-  (%make-atomic-integer value))
+  #+ccl (%make-atomic-reference (make-array 1 :initial-element value))
+  #-ccl (%make-atomic-reference (cons value nil)))
 
 (defmethod print-object ((int atomic-integer) stream)
   (print-unreadable-object (int stream :type t :identity t)
     (format stream "~S" (atomic-get int))))
 
 (defmethod atomic-get ((int atomic-integer))
-  (slot-value int 'cell))
+  (atomic-place int))
 
 (defmethod atomic-cas ((int atomic-integer) expected new)
-  (let* ((*the-int* (slot-value int 'cell))
-         (cas-result (atomics:cas *the-int* expected new)))
-    (when cas-result
-      (setf (slot-value int 'cell) *the-int*))
-    cas-result))
+  (atomics:cas (atomic-place int) expected new))
+
+(defmethod atomic-setf ((int atomic-integer) new)
```

Comment:
OK, this tries until the new value was set.
Could `atomic-cas` also make use of this?

Reply:
`atomic-cas` here is just `cas` special for atomic-reference. I looked up almost all common lisp atomic libs available in Google and found that `atomic-setf` 's a convention to set a place.

Reply:
`atomic-setf` is probably for the case when the 'old' value is ignored. So it's kind of an alternative to `cas` depending on the use-case.

Reply:
> 

For out `cl-gserver` library, yes, as `atomic-setf` is specified to `atomic-reference`, for general usage in other libraries, `atomic-setf` is usually a macro which has such form as `(atomic-setf place old new)`. It shouldn't be a problem as atomic operations are restricted only in several function in `cl-gserver`.